### PR TITLE
Banshee ssr single increment

### PR DESCRIPTION
**Pull request by noah95**
As discussed with @fabianschuiki and @paulsc96 

## Test

Data pointed to by ssr0: `double x[8] = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0};`

In hex
```
3ff0000000000000
4000000000000000
4008000000000000
4010000000000000
4014000000000000
4018000000000000
```

## Before fix

Subsequent reads with more than once the `ft0` as `rsX` results in every other element in `x` being skipped. The pointer is incremented on each call to `banshee_ssr_next` which leads to a double (or more) increment in case of `ft0` being used twice in an instruction.

```
8000006c  f0:3ff0000000000000 f0:4000000000000000 f3=3ff0000000000000  # fsgnj.d ft3, ft0, ft0
80000070  f0:4008000000000000 f0:4010000000000000 f3=4008000000000000  # fsgnj.d ft3, ft0, ft0
80000074  f0:4014000000000000 f0:4018000000000000 f3=4014000000000000  # fsgnj.d ft3, ft0, ft0
```

## After this fix

SSR pointer is only incremented once per program address. This is the correct behaviour.

```
800000a8  f0:3ff0000000000000 f0:3ff0000000000000 f4=3ff0000000000000  # fsgnj.d ft3, ft0, ft0
800000b8  f0:4000000000000000 f0:4000000000000000 f4=4000000000000000  # fsgnj.d ft3, ft0, ft0
800000c8  f0:4008000000000000 f0:4008000000000000 f4=4008000000000000  # fsgnj.d ft3, ft0, ft0
```

### CODE REVIEW

File: sw/banshee/src/tran.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ static NONAME: &'static i8 = unsafe { std::mem::transmute("\0".as_ptr()) };
 
 /// Base address of the stream semantic regsiters
 static SSR_BASE: u64 = 0x204800;
+static SSR_N_STREAMERS: u32 = 2;
```
Comment (SamuelRiedel) [inline sw/banshee/src/tran.rs:26]: Awesome. If you add a new variable for the number of streamers, can I ask you to also update [line 2753](https://github.com/pulp-platform/snitch/blob/b72c089b206785ae0b7ca331d104f52765e134c1/sw/banshee/src/tran.rs#L2753), which seems to have the number of streamers hardcoded to 2. Or am I misunderstanding something?

**Review by SamuelRiedel — CHANGES_REQUESTED** [review]
Very nice thanks. If I understood the SSRs correctly, I have one small request, otherwise, go ahead and merge. :+1:

File: sw/banshee/src/tran.rs (hunk 1 above)
Comment (noah95) [reply sw/banshee/src/tran.rs:26]: That is correct! I looked for other ocurences where `SSR_N_STREAMERS` can be substituted but there might still be some missing though.

**Review by SamuelRiedel — APPROVED** [review]
Nice thanks. This can be merged in my opinion. I think you just need to rustfmt the changes to make the CI happy as well :)

**Review by SamuelRiedel — APPROVED** [review]

**Pull request merged**